src/modules/pages/EditPage/EditApiForm.js

```diff
@@ -1,73 +1,388 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import R from 'ramda';
+import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
-import { reduxForm } from 'redux-form';
+import { Field, formValueSelector, reduxForm } from 'redux-form';
+
+import block from '../../../helpers/bem-cn';
+import transformFormValues from '../../../helpers/transformFormValues';
 
 import Section from '../../Layout/Section/Section';
-import FormRow from '../../forms/FormRow';
-import FormInput from '../../forms/FormInput/FormInput';
-import FormLabel from '../../forms/FormLabel';
+import Title from '../../Layout/Title/Title';
+import Row from '../../Layout/Row/Row';
+import Input from '../../inputs/Input';
+import Radio from '../../inputs/Radio/Radio';
+import Label from '../../labels/Label';
+import Hint from '../../labels/Hint/Hint';
+import MultiSelect from '../../selects/MultiSelect/MultiSelect';
 import Button from '../../buttons/Button';
+import Icon from '../../Icon/Icon';
 
 import RenderPlugins from '../../forms/plugins/RenderPlugins';
 
+const b = block('j-api-form');
+
 const propTypes = {
+    api: PropTypes.object.isRequired,
+    apiSchema: PropTypes.object.isRequired,
+    excludePlugin: PropTypes.func.isRequired,
+    handleDelete: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func.isRequired,
     initialValues: PropTypes.object,
+    location: PropTypes.object,
+    selectPlugin: PropTypes.func.isRequired,
+    selectedPlugins: PropTypes.arrayOf(PropTypes.string).isRequired,
 };
 
 const ApiForm = (props) => {
-    const { handleSubmit } = props;
+    const {
+        api,
+        apiSchema,
+        excludePlugin,
+        initialValues,
+        handleSubmit,
+        plugins,
+        response,
+        selectPlugin,
+        selectedPlugins,
+        location,
+    } = props;
     const parse = value => (value === undefined ? undefined : parseInt(value));
+    const includePlugin = value => {
+        apiSchema.plugins.map((plugin, index) => {
```

Comment:
we could combine filter and map:
```js
apiSchema.plugins.filter((plugin, index) => {
  return plugin.name === value.value && !selectedPlugins.includes(plugin.name)
}).map((plugin, index) => {
  selectPlugin(plugin.name);
});
```

Reply:
Totally agree with you, @rafaeljesus , and ashamed that didn't notice this by myself 😞

Reply:
done